api/src/main/java/graphql/nadel/normalized/ExecutableNormalizedOperationToAstCompiler.java

```diff
@@ -0,0 +1,189 @@
+            if (subSelections.size() > 0) {
+                selectionSet = newSelectionSet()
+                        .selections(subSelections)
+                        .build();
+            }
+            List<Argument> arguments = createArguments(executableNormalizedField, variableAccumulator);
+            Field field = newField()
+                    .name(executableNormalizedField.getFieldName())
+                    .alias(executableNormalizedField.getAlias())
+                    .selectionSet(selectionSet)
+                    .arguments(arguments)
+                    .build();
+
+            groupedFields.computeIfAbsent(objectTypeName, ignored -> new ArrayList<>()).add(field);
+        }
+        return groupedFields;
+    }
+
+    private static SelectionSet selectionSet(List<Field> fields) {
+        return newSelectionSet().selections(fields).build();
+    }
+
+    private static List<Argument> createArguments(ExecutableNormalizedField executableNormalizedField, List<VariableValueWithDefinition> variables) {
+        ImmutableList.Builder<Argument> result = ImmutableList.builder();
+        ImmutableMap<String, NormalizedInputValue> normalizedArguments = executableNormalizedField.getNormalizedArguments();
+        for (String argName : normalizedArguments.keySet()) {
+            NormalizedInputValue normalizedInputValue = normalizedArguments.get(argName);
+            Value<?> value = argValue(normalizedInputValue, variables);
+            Argument argument = newArgument()
+                    .name(argName)
+                    .value(value)
+                    .build();
+            result.add(argument);
+        }
+        return result.build();
+    }
+
+    private static Value<?> argValue(Object value, List<VariableValueWithDefinition> variables) {
+        if (value instanceof List) {
+            ArrayValue.Builder arrayValue = ArrayValue.newArrayValue();
+            arrayValue.values(map((List<Object>) value, val -> argValue(val, variables)));
+            return arrayValue.build();
+        }
+        if (value instanceof Map) {
+            ObjectValue.Builder objectValue = ObjectValue.newObjectValue();
+            Map<String, Object> map = (Map<String, Object>) value;
+            for (String fieldName : map.keySet()) {
+                Value<?> fieldValue = argValue((NormalizedInputValue) map.get(fieldName), variables);
+                objectValue.objectField(ObjectField.newObjectField().name(fieldName).value(fieldValue).build());
+            }
+            return objectValue.build();
+        }
+        if (value == null) {
+            return NullValue.newNullValue().build();
+        }
+        return (Value<?>) value;
+    }
+
+    @NotNull
+    private static Value<?> argValue(NormalizedInputValue normalizedInputValue, List<VariableValueWithDefinition> variableDefinitionAccumulator) {
+        if (JSON_SCALAR_TYPENAME.equals(normalizedInputValue.getUnwrappedTypeName())) {
```

Comment:
Nit: flip this around?

```
normalizedInputValue.getUnwrappedTypeName().equals(JSON_SCALAR_TYPENAME)
```

Reply:
actually I recommend this style as it prevents any NPE from `getUnwrappedTypeName` returning `null`